test/typecheck/main.go

```diff
@@ -62,6 +65,24 @@ var (
 	}
 	darwinPlatString  = "darwin/386,darwin/amd64"
 	windowsPlatString = "windows/386,windows/amd64"
+
+	// directories we always ignore
+	standardIgnoreDirs = []string{
+		// Staging code is symlinked from vendor/k8s.io, and uses import
+		// paths as if it were inside of vendor/. It fails typechecking
+		// inside of staging/, but works when typechecked as part of vendor/.
+		"staging",
+		// OS-specific vendor code tends to be imported by OS-specific
+		// packages. We recursively typecheck imported vendored packages for
+		// each OS, but don't typecheck everything for every OS.
+		"vendor",
```

Comment:
this isn't new, but I was slightly surprised by it... I think this means if we add a main package under vendor or staging, typecheck won't check its import tree. Worth remembering as we try to push more things out of root k/k

Reply:
we still typecheck this IIUC, as in, we typecheck files which then import vendor, which transitiviely checks.

we don't typecheck literally all code in vendor though.